operator/pkg/component/component/component.go

```diff
@@ -253,382 +159,43 @@ func isMeshConfigMap(obj *object.K8sObject) bool {
-	cn := name.GalleyComponentName
-	return &GalleyComponent{
-		&CommonComponentFields{
-			Options:       opts,
-			componentName: cn,
-		},
-	}
-}
-
-// Run implements the IstioComponent interface.
-func (c *GalleyComponent) Run() error {
-	return runComponent(c.CommonComponentFields)
-}
-
-// RenderManifest implements the IstioComponent interface.
-func (c *GalleyComponent) RenderManifest() (string, error) {
-	if !c.started {
-		return "", fmt.Errorf("component %s not started in RenderManifest", c.ComponentName())
-	}
-	return renderManifest(c.CommonComponentFields)
-}
-
-// ComponentName implements the IstioComponent interface.
-func (c *GalleyComponent) ComponentName() name.ComponentName {
-	return c.CommonComponentFields.componentName
-}
-
-// ResourceName implements the IstioComponent interface.
-func (c *GalleyComponent) ResourceName() string {
-	return c.CommonComponentFields.resourceName
-}
-
-// Namespace implements the IstioComponent interface.
-func (c *GalleyComponent) Namespace() string {
-	return c.CommonComponentFields.Namespace
-}
-
-// CNIComponent is the egress gateway component.
-type CNIComponent struct {
-	*CommonComponentFields
-}
-
-// NewCNIComponent creates a new IngressComponent and returns a pointer to it.
-func NewCNIComponent(opts *Options) *CNIComponent {
-	cn := name.CNIComponentName
-	return &CNIComponent{
-		&CommonComponentFields{
-			Options:       opts,
-			componentName: cn,
-		},
-	}
-}
-
-// Run implements the IstioComponent interface.
-func (c *CNIComponent) Run() error {
-	return runComponent(c.CommonComponentFields)
-}
-
-// RenderManifest implements the IstioComponent interface.
-func (c *CNIComponent) RenderManifest() (string, error) {
-	if !c.started {
-		return "", fmt.Errorf("component %s not started in RenderManifest", c.ComponentName())
-	}
-	return renderManifest(c.CommonComponentFields)
-}
-
-// ComponentName implements the IstioComponent interface.
-func (c *CNIComponent) ComponentName() name.ComponentName {
-	return c.CommonComponentFields.componentName
-}
-
-// ResourceName implements the IstioComponent interface.
-func (c *CNIComponent) ResourceName() string {
-	return c.CommonComponentFields.resourceName
-}
-
-// Namespace implements the IstioComponent interface.
-func (c *CNIComponent) Namespace() string {
-	return c.CommonComponentFields.Namespace
-}
-
-// IngressComponent is the ingress gateway component.
-type IngressComponent struct {
-	*CommonComponentFields
-}
-
 // NewIngressComponent creates a new IngressComponent and returns a pointer to it.
-func NewIngressComponent(resourceName string, index int, opts *Options) *IngressComponent {
-	cn := name.IngressComponentName
-	return &IngressComponent{
-		CommonComponentFields: &CommonComponentFields{
-			Options:       opts,
-			componentName: cn,
-			resourceName:  resourceName,
-			index:         index,
-		},
+func NewIngressComponent(resourceName string, index int, opts *Options) IstioComponent {
```

Comment:
i think we still need the interface methods implemented for these, because controlplane has a single list of components, including gateways and addons.

Reply:
not sure what you mean? this returns a normal IstioComponentImpl struct, just with the additional info needed. I updated the methods to take these into account